lib/validateMessage.js

```diff
@@ -63,6 +63,21 @@ exports.validateMessage = function validateMessage(raw) {
     var type = match[3];
     var scope = match[4];
     var subject = match[5];
+    config.scopes = config.scopes || {};
+    var validateScopes = config.scopes.validate || false;
+    var multipleScopesAllowed = config.scopes.multiple || false;
+    var allowedScopes = config.scopes.allowed || '*';
+    var scopeRequired = config.scopes.required || false;
+    var scopes = scope ? scope.split(',') : [];
+    var validateScope = (item) => {
```

Comment:
We are not ready yet I believe (correct me if I am wrong @kentcdodds) to drop support for `0.10.x`. You could refactor this to es5 function?

Another idea, as this function is getting too big, would be to split all these validations. We can start by splitting the scope `if (validateScopes) {...}` validation and move it to `lib/validation.js` which could export validation functions? This should make the code easier to test in isolation as well. What do you think?